File: DSA/Vikash_kumar/Day_9_Question_2.py
#Day9-Question2-
def is_balanced(brackets):
    stack = []
    bracket_map = {')': '(', '}': '{', ']': '['}

    for char in brackets:
        if char in bracket_map.values():
            stack.append(char)
        elif char in bracket_map.keys():
            if stack == [] or bracket_map[char] != stack.pop():
                return "NO"
        else:
            continue

    return "YES" if stack == [] else "NO"

string = input("Enter the string : ")
print(is_balanced(string))

File: DSA/Manav-Rajput/Day9_question_2.java
import java.util.Stack;

public class BalancedBrackets {

    public static String isBalanced(String brackets) {
        Stack<Character> stack = new Stack<>();
        
        for (char ch : brackets.toCharArray()) {
            if (ch == '(' || ch == '{' || ch == '[') {
                stack.push(ch);
            } else if (ch == ')' || ch == '}' || ch == ']') {
                if (stack.isEmpty()) {
                    return "NO";
                }
                char top = stack.pop();
                if ((ch == ')' && top != '(') ||
                    (ch == '}' && top != '{') ||
                    (ch == ']' && top != '[')) {
                    return "NO";
                }
            }
        }
        
        return stack.isEmpty() ? "YES" : "NO";
    }

    public static void main(String[] args) {
        String string1 = "{[()]}";
        String string2 = "{[(])}";
        
        System.out.println(isBalanced(string1)); // Output: YES
        System.out.println(isBalanced(string2)); // Output: NO
    }
}
